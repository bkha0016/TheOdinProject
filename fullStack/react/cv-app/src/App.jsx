import { useState } from "react";
import GeneralInfo from "./components/GeneralInfo.jsx";
import EducationalExperience from "./components/EducationalExperience.jsx";
import PracticalExperience from "./components/PracticalExperience.jsx";
import "./styles/App.css";

function App() {
  return (
    <div className="App">
      <h1>CV Application</h1>
      <GeneralInfo />
      <EducationalExperience />
      <PracticalExperience />
    </div>
  );
}

export default App;
