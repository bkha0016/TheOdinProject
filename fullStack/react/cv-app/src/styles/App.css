body {
  font-family: Arial, sans-serif;
  background-color: #f4f4f4;
  margin: 0;
  padding: 20px;
}

.cv-container {
  max-width: 800px;
  margin: 0 auto;
  background-color: #ffffff;
  padding: 20px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  border-radius: 8px;
}

h1 {
  text-align: center;
  color: #333;
}

h2 {
  border-bottom: 2px solid #333;
  padding-bottom: 5px;
  margin-bottom: 15px;
}

section {
  margin-bottom: 30px;
}

button {
  background-color: #4caf50;
  color: white;
  border: none;
  padding: 10px 20px;
  margin: 10px 0;
  border-radius: 5px;
  cursor: pointer;
}

button:hover {
  background-color: #45a049;
}

input,
textarea {
  width: 100%;
  padding: 10px;
  margin: 5px 0;
  border: 1px solid #ccc;
  border-radius: 5px;
}

input[type="text"],
input[type="email"],
input[type="tel"] {
  height: 40px;
}

textarea {
  resize: none;
}

.cv-section {
  margin-bottom: 20px;
}

.cv-section h2 {
  font-size: 1.2em;
}

.cv-section p {
  margin: 5px 0;
}

.cv-section p span {
  font-weight: bold;
}

.add-more-button {
  background-color: #007bff;
}

.add-more-button:hover {
  background-color: #0056b3;
}

#general-info,
#education,
#practical-experience {
  border-bottom: 1px solid #ddd;
  padding-bottom: 15px;
  margin-bottom: 15px;
}

#general-info h2,
#education h2,
#practical-experience h2 {
  color: #444;
}

#general-info p,
#education p,
#practical-experience p {
  line-height: 1.6;
}

#education .education-entry,
#practical-experience .experience-entry {
  margin-top: 10px;
  padding-left: 10px;
  border-left: 2px solid #ddd;
}

.cv-result h2 {
  color: #666;
}

.cv-result p {
  margin: 10px 0;
}
