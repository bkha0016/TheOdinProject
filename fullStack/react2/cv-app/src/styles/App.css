.app {
  max-width: 600px;
  margin: 0 auto;
  padding: 20px;
  text-align: center;
}

input,
button {
  margin: 10px 0;
  padding: 10px;
  width: 80%;
  font-size: 16px;
}

button {
  cursor: pointer;
  background-color: #4caf50;
  color: white;
  border: none;
  border-radius: 5px;
}

.preview {
  margin-top: 30px;
  padding: 20px;
  border: 1px solid #ddd;
  border-radius: 5px;
  text-align: left;
}

.preview h2 {
  margin-bottom: 20px;
}
