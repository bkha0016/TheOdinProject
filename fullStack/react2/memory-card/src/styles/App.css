.app {
  font-family: "Segoe UI", sans-serif;
  text-align: center;
  padding: 2rem;
  background-color: #f0f4f8;
  min-height: 100vh;
}

h1 {
  font-size: 2.5rem;
  color: #333;
  margin-bottom: 1.5rem;
}
