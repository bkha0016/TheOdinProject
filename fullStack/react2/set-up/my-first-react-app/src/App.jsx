import React, { useState } from "react";

function App() {
  const [message, setMessage] = useState("Hello, World!");
  const [heading, setHeading] = useState("Magnificent Monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos");
  };

  const changeMessage = () => {
    setMessage((prev) =>
      prev === "Hello, World!" ? "Welcome to React!" : "Hello, World!"
    );
  };

  return (
    <div style={{ textAlign: "center", marginTop: "50px" }}>
      <h1>test</h1>
      <h1>{message}</h1> {/* Display the dynamic message */}
      <button
        onClick={changeMessage}
        style={{
          padding: "10px 20px",
          fontSize: "16px",
          backgroundColor: "#4CAF50",
          color: "white",
          border: "none",
          borderRadius: "5px",
          cursor: "pointer",
        }}
      >
        Click Me!
      </button>
      <button type="button" onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </div>
  );
}

export default App;
