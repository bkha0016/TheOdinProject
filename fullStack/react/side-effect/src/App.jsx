import { useState, useEffect } from "react";
import "./App.css";

function Clock() {
  const [counter, setCounter] = useState(0);

  useEffect(() => {
    const key = setInterval(() => {
      setCounter((count) => count + 1);
    }, 1000);

    return () => {
      clearInterval(key);
    };
  }, []);

  return <p>{counter} seconds have passed.</p>;
}

function Testinput() {
  const [input, setInput] = useState("");

  const handleInput = (e) => {
    setInput(e.target.value);
  };

  return (
    <>
      <input onChange={handleInput} value={input}></input>
      <p>{input}</p>
    </>
  );
}

function App() {
  const [count, setCount] = useState(0);

  return (
    <>
      <Clock />
      <Testinput />
    </>
  );
}

export default App;
