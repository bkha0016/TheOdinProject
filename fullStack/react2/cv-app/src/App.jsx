import React, { useState } from "react";
import GeneralInfo from "./components/GeneralInfo";
import EducationalExperience from "./components/EducationalExperience";
import PracticalExperience from "./components/PracticalExperience";
import "./styles/App.css";

export default function App() {
  const [generalInfo, setGeneralInfo] = useState({
    name: "",
    email: "",
    phone: "",
  });

  const [education, setEducation] = useState([
    { school: "", title: "", date: "" },
  ]);

  const [experience, setExperience] = useState([
    { company: "", position: "", responsibilities: "", from: "", until: "" },
  ]);

  return (
    <div className="app">
      <h1>Create Your CV</h1>

      {/* General Info Section */}
      <GeneralInfo
        data={generalInfo}
        onChange={(updatedData) => setGeneralInfo(updatedData)}
      />

      {/* Educational Experience Section */}
      <EducationalExperience
        data={education}
        onChange={(updatedData) => setEducation(updatedData)}
      />

      {/* Practical Experience Section */}
      <PracticalExperience
        data={experience}
        onChange={(updatedData) => setExperience(updatedData)}
      />

      {/* Preview Section */}
      <div className="preview">
        <h2>Preview of Your CV</h2>
        <p>
          <strong>Name:</strong> {generalInfo.name}
        </p>
        <p>
          <strong>Email:</strong> {generalInfo.email}
        </p>
        <p>
          <strong>Phone:</strong> {generalInfo.phone}
        </p>

        <h3>Education</h3>
        {education.map((entry, index) => (
          <div key={index}>
            <p>
              <strong>School:</strong> {entry.school}
            </p>
            <p>
              <strong>Title:</strong> {entry.title}
            </p>
            <p>
              <strong>Date:</strong> {entry.date}
            </p>
          </div>
        ))}

        <h3>Practical Experience</h3>
        {experience.map((entry, index) => (
          <div key={index}>
            <p>
              <strong>Company:</strong> {entry.company}
            </p>
            <p>
              <strong>Position:</strong> {entry.position}
            </p>
            <p>
              <strong>Responsibilities:</strong> {entry.responsibilities}
            </p>
            <p>
              <strong>From:</strong> {entry.from}
            </p>
            <p>
              <strong>Until:</strong> {entry.until}
            </p>
          </div>
        ))}
      </div>

      <button
        onClick={() => downloadCVData({ generalInfo, education, experience })}
      >
        Export CV
      </button>
    </div>
  );
}

function downloadCVData(data) {
  const blob = new Blob([JSON.stringify(data, null, 2)], {
    type: "application/json",
  });
  const url = URL.createObjectURL(blob);
  const a = document.createElement("a");
  a.href = url;
  a.download = "cvData.json";
  document.body.appendChild(a);
  a.click();
  document.body.removeChild(a);
}
