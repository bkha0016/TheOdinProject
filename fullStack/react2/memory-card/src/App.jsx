import { useEffect, useState } from "react";
import CardGrid from "./components/CardGrid";
import Scoreboard from "./components/Scoreboard";
import useFetchData from "./hooks/useFetchData";
import "./styles/App.css";

export default function App() {
  const { data: cards, loading, error } = useFetchData();
  const [shuffledCards, setShuffledCards] = useState([]);
  const [clickedIds, setClickedIds] = useState(new Set());
  const [score, setScore] = useState(0);
  const [bestScore, setBestScore] = useState(0);

  useEffect(() => {
    if (cards.length) {
      setShuffledCards(shuffleArray(cards));
    }
  }, [cards]);

  function shuffleArray(array) {
    const shuffled = [...array];

    for (let i = shuffled.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [shuffled[i], shuffled[j]] = [shuffled[j], shuffled[i]];
    }
    return shuffled;
  }

  // function shuffleAndSetCards(cardArray) {
  //   const shuffled = [...cardArray].sort(() => 0.5 - Math.random());
  //   setShuffledCards(shuffled);
  // }

  function handleCardClick(id) {
    if (clickedIds.has(id)) {
      setScore(0);
      setClickedIds(new Set()); // Reset
    } else {
      const newClickedIds = new Set(clickedIds);
      newClickedIds.add(id);
      setClickedIds(newClickedIds);

      const newScore = score + 1;
      setScore(newScore);
      setBestScore((prev) => Math.max(prev, newScore));
    }

    // Shuffle after every click
    setShuffledCards(shuffleArray(cards));
  }
  // if (loading) return <p>Loading...</p>;
  // if (error) return <p>Error loading Cards: {error}</p>;

  return (
    <div className="app">
      <h1>Memory Card Game</h1>
      <Scoreboard currentScore={score} bestScore={bestScore} />
      {loading && <p>Loading cards...</p>}
      {error && <p>{error}</p>}
      {!loading && !error && (
        <CardGrid cards={shuffledCards} onCardClick={handleCardClick} />
      )}
    </div>
  );
}
