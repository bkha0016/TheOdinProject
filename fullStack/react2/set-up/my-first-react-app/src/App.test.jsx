import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

// describe("App component", () => {
//   it("renders correct heading", () => {
//     render(<App />);
//     expect(screen.getByRole("heading").textContent).toMatch(/test/i);
//   });
// });

describe("App component", () => {
  it("renders magnificent monkeys", () => {
    // since screen does not have the container property, we'll destructure
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();
    render(<App />);
    const button = screen.getByRole("button", { name: "Click Me" });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
  });
});

describe("something truthy and falsy", () => {
  it("true to be true", () => {
    expect(true).toBe(true);
  });

  it("false to be false", () => {
    expect(false).toBe(false);
  });
});
