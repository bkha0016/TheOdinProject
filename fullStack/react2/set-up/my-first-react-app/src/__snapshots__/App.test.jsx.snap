// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders magnificent monkeys 1`] = `
<div>
  <div
    style="text-align: center; margin-top: 50px;"
  >
    <h1>
      test
    </h1>
    <h1>
      Hello, World!
    </h1>
     
    <button
      style="padding: 10px 20px; font-size: 16px; background-color: rgb(76, 175, 80); color: white; border-radius: 5px; cursor: pointer;"
    >
      Click Me!
    </button>
    <button
      type="button"
    >
      Click Me
    </button>
    <h1>
      Magnificent Monkeys
    </h1>
  </div>
</div>
`;
